pkg/ext/delegate_test.go

```diff
@@ -0,0 +1,1012 @@
+package ext
```

Comment:
Can we move the test cases here to `delegate_error_test.go` so that we test both delegate at once? This would also ensure that we haven't forgotten a call to `convertError`.

eg:

```go

			testDelegate := &delegate[*TestType, *TestTypeList]{
				scheme: scheme,
				t:      &TestType{},
				tList:  &TestTypeList{},
				gvk:    gvk,
				gvr:    gvr,
				store:  mockStore,
			}
```

becomes

```go

			testDelegate := &delegateError{
                           inner: &delegate[*TestType, *TestTypeList]{
				scheme: scheme,
				t:      &TestType{},
				tList:  &TestTypeList{},
				gvk:    gvk,
				gvr:    gvr,
				store:  mockStore,
                          }
			}
```

And then just run your tests on this `testDelegate` as you're doing here (and ensure you're getting internal errors when appropriate.